Client/Frontend/Settings/SearchBar/SearchBarSettingsViewModel.swift

```diff
@@ -0,0 +1,97 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+
+enum SearchBarPosition: String {
+    case bottom
+    case top
+
+    var getLocalizedTitle: String {
+        switch self {
+        case .bottom:
+            return .Settings.Toolbar.Bottom
+        case .top:
+            return .Settings.Toolbar.Top
+        }
+    }
+}
+
+protocol SearchBarPreferenceDelegate: AnyObject {
+    func didUpdateSearchBarPositionPreference()
+}
+
+final class SearchBarSettingsViewModel {
+
+    var title: String = .Settings.Toolbar.Toolbar
+    weak var delegate: SearchBarPreferenceDelegate?
+
+    private let prefs: Prefs
+    init(prefs: Prefs) {
+        self.prefs = prefs
+    }
+
+    static var isEnabled: Bool {
+        return UIDevice.current.userInterfaceIdiom != .pad
+    }
+
+    var searchBarTitle: String {
+        searchBarPosition.getLocalizedTitle
+    }
+
+    var searchBarPosition: SearchBarPosition {
+        let defaultPosition = getDefaultSearchPosition()
+        guard let raw = prefs.stringForKey(PrefsKeys.KeySearchBarPosition) else {
+            saveSearchBarPosition(defaultPosition)
+            return defaultPosition
+        }
+
+        let position = SearchBarPosition(rawValue: raw) ?? .bottom
+        return position
+    }
+
+    var topSetting: CheckmarkSetting {
+        return CheckmarkSetting(
+            title: NSAttributedString(string: SearchBarPosition.top.getLocalizedTitle), subtitle: nil,
+            accessibilityIdentifier: AccessibilityIdentifiers.Settings.SearchBar.topSetting,
+            isChecked: { return self.searchBarPosition == .top },
+            onChecked: { self.saveSearchBarPosition(SearchBarPosition.top)}
+        )
+    }
```

Comment:
```suggestion
    var topSetting: CheckmarkSetting {
        return CheckmarkSetting( title: NSAttributedString(string: SearchBarPosition.top.getLocalizedTitle), 
            subtitle: nil,      
            accessibilityIdentifier: AccessibilityIdentifiers.Settings.SearchBar.topSetting,
            isChecked: { return self.searchBarPosition == .top },
            onChecked: { self.saveSearchBarPosition(SearchBarPosition.top)}
        )
    }
```

nit: better indentation

Reply:
nit: somehow indentation looked off

Reply:
Ah, it's an indentation type where we put the first parameter on a new line. I guess that's something we can discuss in our style guide.

Reply:
I changed it to something more standard in the meantime